meta-balena-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -165,6 +165,11 @@ if command -v "systemctl"; then
     done
 fi
 
+if [ -z "${INTERNAL_DEVICE_KERNEL}" ]; then
+    report_progress 100 "Installer is not supported on this device type."
+    fail "Undefined target programming device - make sure INTERNAL_DEVICE_KERNEL is defined"
+fi
```

Comment:
I'm not sure I understand why we can't have a build time check for this, that only fails on flasher device types if the var is unset.

Reply:
I guess I could check the contracts to distinguish flasher from non-flasher devices. I'll add that.